### PR TITLE
Feature/Toggle accounts visibility from accounts page

### PR DESCRIPTION
**Pull request by bptrgx**
When I want to toggle the visibility of an account, going through the create-or-update-account-dialog is too long (_burger menu -> edit -> "Exclude from Analysis" -> Save_). The visibility indicator (the crossed eye) should be a button that toggles the visibility of the account.

Now:

With this PR:

### CODE REVIEW

File: apps/client/src/app/components/accounts-table/accounts-table.component.html [hunk 1]
```diff
@@ -5,9 +5,23 @@
...
-    <td *matCellDef="let element" class="d-none d-lg-table-cell px-1" mat-cell>
+    <td
+      *matCellDef="let element"
+      class="d-none d-lg-table-cell px-1"
+      mat-cell
+      (click)="onToggleExcluded(element); $event.stopPropagation()"
+    >
       <div class="d-flex justify-content-center">
-        <ion-icon *ngIf="element.isExcluded" name="eye-off-outline"></ion-icon>
+        <ion-icon
```
Comment (dtslvr) [inline apps/client/src/app/components/accounts-table/accounts-table.component.html:15]: Can you wrap the icon into a button?

```
<button
  class="mx-1 no-min-width px-2"
  mat-button
  (click)="onToggleExcluded(element); $event.stopPropagation()"
>
  <ion-icon [name]="element.isExcluded ? 'eye-off-outline' : 'eye-outline'" [ngClass]="{ 'text-muted': !element.isExcluded }"></ion-icon>
</button>
```

File: apps/client/src/app/components/accounts-table/accounts-table.component.scss [hunk 2]
```diff
@@ -4,6 +4,15 @@
   display: block;
 
   .mat-mdc-table {
+    #isExcluded-icon {
```
Comment (dtslvr) [inline apps/client/src/app/components/accounts-table/accounts-table.component.scss:7]: We don't use ids for styling. With my suggestion above, you don't need it at all.

File: apps/client/src/app/components/accounts-table/accounts-table.component.ts [hunk 3]
```diff
@@ -87,6 +88,10 @@ export class AccountsTableComponent implements OnChanges, OnDestroy, OnInit {
     }
   }
 
+  public onToggleExcluded(aAccount: AccountModel) {
```
Comment (dtslvr) [inline apps/client/src/app/components/accounts-table/accounts-table.component.ts:91]: Can you move it down to keep the alphabetical order?

File: apps/client/src/app/pages/accounts/accounts-page.component.ts [hunk 4]
```diff
@@ -144,6 +144,32 @@ export class AccountsPageComponent implements OnDestroy, OnInit {
       });
   }
 
+  public onToggleExcluded(aAccount: AccountModel) {
```
Comment (dtslvr) [inline apps/client/src/app/pages/accounts/accounts-page.component.ts:147]: `onToggleExcludedAccount({accountType, balance, comment, ...}: AccountModel)`

File: apps/client/src/app/pages/accounts/accounts-page.component.ts [hunk 5]
```diff
@@ -144,6 +144,32 @@ export class AccountsPageComponent implements OnDestroy, OnInit {
...
+      balance: aAccount.balance,
+      comment: aAccount.comment,
+      currency: aAccount.currency,
+      id: aAccount.id,
+      isExcluded: !aAccount.isExcluded,
+      name: aAccount.name,
+      platformId: aAccount.platformId
+    };
+    this.dataService
+      .putAccount(account)
```
Comment (dtslvr) [inline apps/client/src/app/pages/accounts/accounts-page.component.ts:159]:
```
.putAccount({
  accountType,
  balance,
  comment,
  ...
})
```

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thanks for your contribution @bptrgx. I have added some comments.

**Comment by bptrgx** [comment]
Hi @dtslvr, I've made the changes you suggested.

**Pull request closed without merging**